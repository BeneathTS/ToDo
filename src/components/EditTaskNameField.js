import React, { Component } from 'react';

export default class EditTaskNameField extends Component{
	submitEditedTaskName = (event) => {
		event.preventDefault();
		const oldTaskName = this.props.taskName;
		const newTaskName = event.type === "submit" ?
			event.target.newTaskName.value :
			event.target.value;

		this.props.changeTaskName(oldTaskName, newTaskName);
	}

	componentDidMount() {
		const editTaskNameField = document.getElementById("editTaskNameField");

		editTaskNameField.focus();
		editTaskNameField.select();
	}

	render() {
		return (
			<form onSubmit={this.submitEditedTaskName}>
				<input
					type="text"
					id="editTaskNameField"
					name="newTaskName"
					defaultValue={this.props.taskName}
					onBlur={this.submitEditedTaskName}
				/>
			</form>
		);
	}
}