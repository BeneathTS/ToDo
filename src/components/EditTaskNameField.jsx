import React, { Component, createRef } from 'react';
import { PropTypes } from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import {
  changeTaskNameOnSubmitAction,
  changeTaskNameOnBlurAction,
} from '../actions/actions';

const putStoreToEditTaskNameField = ({ tasks }) => ({ tasks });

const putActionsToEditTaskNameField = (dispatch) => ({
  changeTaskNameOnBlur: bindActionCreators(changeTaskNameOnBlurAction, dispatch),
  changeTaskNameOnSubmit: bindActionCreators(changeTaskNameOnSubmitAction, dispatch),
});

class EditTaskNameField extends Component {
  constructor(props) {
    super(props);
    this.editTaskNameField = createRef();
  }

  componentDidMount() {
    this.editTaskNameField.current.focus();
    this.editTaskNameField.current.select();
  }

  submitEditedTaskNameBySubmit = (event) => {
    event.preventDefault();

    const { changeTaskNameOnSubmit, toggleEditMode } = this.props;
    const { id } = event.target.parentNode;
    const { value: newTaskName } = event.target.editTaskNameField;

    changeTaskNameOnSubmit(id, newTaskName);
    toggleEditMode(event);
  }

  submitEditedTaskNameByBlur = (event) => {
    const { changeTaskNameOnBlur, toggleEditMode } = this.props;
    const { id } = event.target.parentNode.parentNode;
    const { value: newTaskName } = event.target;

    changeTaskNameOnBlur(id, newTaskName);
    toggleEditMode(event);
  }

  render() {
    const { taskName } = this.props;
    return (
      <form onSubmit={this.submitEditedTaskNameBySubmit}>
        <input
          ref={this.editTaskNameField}
          type="text"
          id="editTaskNameField"
          name="editTaskNameField"
          defaultValue={taskName}
          onBlur={this.submitEditedTaskNameByBlur}
        />
      </form>
    );
  }
}

EditTaskNameField.propTypes = {
  taskName: PropTypes.string.isRequired,
  changeTaskNameOnSubmit: PropTypes.func.isRequired,
  changeTaskNameOnBlur: PropTypes.func.isRequired,
  toggleEditMode: PropTypes.func.isRequired,
};

export default connect(
  putStoreToEditTaskNameField,
  putActionsToEditTaskNameField,
)(EditTaskNameField);
