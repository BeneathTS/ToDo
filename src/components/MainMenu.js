import React from 'react';
import '../styles/MainMenu.css';

function UserProfileImage() {
	return (
		<div className="profileImageStub"/>
);
}

function UserProfileName() {
	return (
		<div className="profileNameStub"/>
	);
}

function UserProfileInfo() {
	return (
		<div className="userProfileInfo">
			<UserProfileImage />
			<UserProfileName />
		</div>
	);
}

function TaskLists() {
	return (
		<ul className="taskList">
			<li id="mainTaskList" className="selected">
				<span>Today</span>
			</li>
		</ul>
	);
}

export default function MainMenu () {
	return (
		<div className="mainMenu">
			{/* <UserProfileInfo /> */}
			{/* <TaskLists /> */}
		</div>
	);
}
