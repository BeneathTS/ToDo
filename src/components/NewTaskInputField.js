import React from 'react';
import '../styles/NewTaskInputField.css';

export default class NewTaskIputField extends React.Component {
	render() {
		return (
			<form className="inputField">
				<input type="text" id="newTaskInputField" placeholder="Что будем делать?" />
			</form>
		);
	}

}