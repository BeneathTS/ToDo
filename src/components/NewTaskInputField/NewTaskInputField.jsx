import React from 'react';
import { PropTypes } from 'prop-types';
import { inputField, newTaskInputField} from './NewTaskInputField.module.css';

export default class NewTaskInputField extends React.Component {
  placeholder = 'What we gonna do?';

  submitNewTask = (event) => {
    event.preventDefault();

    const { addSubmittedTask } = this.props;
    const { newTaskInputField } = event.target;
    const submittedTask = newTaskInputField.value;
    if (!submittedTask) return;

    addSubmittedTask(submittedTask);
    newTaskInputField.value = '';
  }

  render() {
    return (
      <form className={inputField} onSubmit={this.submitNewTask}>
        <input
          type="text"
          className={newTaskInputField}
          name="newTaskInputField"
          placeholder={this.placeholder}
        />
      </form>
    );
  }
}

NewTaskInputField.propTypes = {
  addSubmittedTask: PropTypes.func.isRequired,
};
