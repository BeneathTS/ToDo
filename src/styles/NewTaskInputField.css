.inputField {
	width: 100%;
	height: 35px;
}

.inputField input[id="newTaskInputField"]{
	transition: 0.3s;
	width: 100%;
	height: 33px;
	border: 0;
	margin: 0;
	outline: 0;
	font-family: sans-serif;
	font-size: 18px;
	text-align: center;
	color: #212121;
	background-color: #2b2b2b;
}

.inputField input[id="newTaskInputField"]:focus::placeholder{
	transition: 0.8s;
	color: #222;
}

.inputField input[id="newTaskInputField"]:focus{
	transition: 0.8s;
	background-color: #888;
}