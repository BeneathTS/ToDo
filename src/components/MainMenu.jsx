import React from 'react';
import '../styles/MainMenu.css';

// const UserProfileImage = () =>
//  <div className="profileImageStub"/>;

// const UserProfileName =() =>
//  <div className="profileNameStub"/>;

// function UserProfileInfo() {
//  return (
//    <div className="userProfileInfo">
//      <UserProfileImage />
//      <UserProfileName />
//    </div>
//  );
// }

// function TaskLists() {
//  return (
//    <ul className="taskList">
//      <li id="mainTaskList" className="selected">
//        <span>Today</span>
//      </li>
//    </ul>
//  );
// }

export default function MainMenu() {
  return (
    <div className="mainMenu">
      {/* <UserProfileInfo /> */}
      {/* <TaskLists /> */}
    </div>
  );
}
