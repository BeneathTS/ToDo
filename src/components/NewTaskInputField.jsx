import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { bindActionCreators } from 'redux';
import { addSubmittedTaskAction } from '../actions/actions';
import styles from '../styles/NewTaskInputField.module.css';

const putStoreToNewTaskInputField = ({ tasks }) => ({ tasks });

const putActionsToNewTaskInputField = (dispatch) => ({
  addSubmittedTask: bindActionCreators(addSubmittedTaskAction, dispatch),
});

export class NewTaskInputField extends React.Component {
  placeholder = 'What we gonna do?';

  submitNewTask = (event) => {
    event.preventDefault();

    const { addSubmittedTask } = this.props;
    const { newTaskInputField } = event.target;
    const submittedTask = newTaskInputField.value;
    if (!submittedTask) return;

    addSubmittedTask(submittedTask);
    newTaskInputField.value = '';
  }

  render() {
    return (
      <form className={styles.inputField} onSubmit={this.submitNewTask}>
        <input
          type="text"
          id="newTaskInputField"
          name="newTaskInputField"
          placeholder={this.placeholder}
        />
      </form>
    );
  }
}

NewTaskInputField.propTypes = {
  addSubmittedTaskName: PropTypes.func.isRequired,
};

export default connect(
  putStoreToNewTaskInputField,
  putActionsToNewTaskInputField,
)(NewTaskInputField);
