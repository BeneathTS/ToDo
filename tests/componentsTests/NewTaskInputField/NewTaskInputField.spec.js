import React from 'react';
import { NewTaskInputField } from '../../../src/components/NewTaskInputField';
import renderer from 'react-test-renderer';
// import { shallow } from 'enzyme';

describe('NewTaskInputField tests', () => {
  it ('Snapshot check', () => {
    const mockFn = jest.fn();
    const NewTaskInputFieldComponent = renderer.create(<NewTaskInputField addSubmittedTaskName={mockFn} />).toJSON();

    expect(NewTaskInputFieldComponent).toMatchSnapshot();
  });

});