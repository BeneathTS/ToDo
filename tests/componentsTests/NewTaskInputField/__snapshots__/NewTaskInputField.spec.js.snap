// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTaskInputField tests Snapshot check 1`] = `
<form
  className="inputField"
  onSubmit={[Function]}
>
  <input
    id="newTaskInputField"
    name="newTaskInputField"
    placeholder="What we gonna do?"
    type="text"
  />
</form>
`;
